subsys/shell/shell_telnet.c

```diff
@@ -245,6 +245,10 @@ static void telnet_accept(struct net_context *client,
 
 	sh_telnet->client_ctx = client;
 
+	/* Save information about remote. */
```

Comment:
This is in wrong place. It is not telnet shell's responsibility to set something like this. For non-offloading case, the remote value is set by `NET_CONN_CB(tcp_syn_rcvd)` in net_context.c when the connection is established.

Reply:
Yes, this change was somewhat "iffy" to me also.. But I did it here instead of the offloading driver because it felt somewhat "generic".. 

I agree that it makes more sense for the offloading driver to fill out this info.. Will update..